packages/core/contracts/oracle/implementation/VotingV2.sol

```diff
@@ -886,29 +879,13 @@ contract VotingV2 is
                 // must have been rolled. In this case, update the internal trackers for this vote.
                 if (priceRequest.lastVotingRound < currentRoundId) {
                     priceRequest.lastVotingRound = SafeCast.toUint32(currentRoundId);
-                    // If the lastRolledIndex is not set or the last rolled index is not equal to the previous index
-                    // then set it to the current index. This acts to only update hte lastRolledIndex when we transition
-                    // over ranges of continuous rolled votes.
-                    if (
-                        lastRolledIndex == 0 ||
-                        (lastRolledIndex != 0 && skippedRequestIndexes[lastRolledIndex] != requestIndex - 1)
-                    ) lastRolledIndex = requestIndex;
-                    // Set the skippedRequestIndexes mapping at the last rolled index to the current index. This acts
-                    // to create a jump from the first rolled index (last rollIndex was no updated if sequential) to the
-                    // current request index, thereby making a continuous jump over all rolled votes.
-                    skippedRequestIndexes[lastRolledIndex] = requestIndex;
-                    // If the lastRolled index is not the current index (multi-round roll) then set the requestIndex equal
-                    // to its self to indicate that this index was jumped. This will never normally be hit within round
-                    // traversal (due to jumping over a range of rolled votes) but it is useful to have to see that a
-                    // vote was skipped. This is also levered in the logic outside of the forloop for other slashing.
-                    if (lastRolledIndex != requestIndex) skippedRequestIndexes[requestIndex] = requestIndex;
+
+                    // This is a subtle operation. This is not setting the skip value for the _current request_ to this
+                    // value. It is setting the skip value for the element after the last processed index to the skip
+                    // value. This causes this skip interval to extend on each subsequent rolled request because no
+                    // new elements are processed on a skip, thereby leaving nextIndexToProcess the same.
+                    skippedRequestIndexes[nextIndexToProcess] = requestIndex;
```

Comment:
@mrice32 this is very cleaver. I did not fully appreciate how this works until now.